Svc/TraceFileLogger/TraceFileLogger.cpp

```diff
@@ -0,0 +1,242 @@
+            FwSignedSizeType fileSize;
+            (void) this->m_log_file.position(fileSize);
+            Os::File::Status stat = this->m_log_file.write(reinterpret_cast<const U8*>(data),writeSize,Os::File::WAIT);
+            // Assert if file is not already open
+            FW_ASSERT(stat != Os::File::NOT_OPENED);
+            this->m_byteCount += (size);
+        }
+    }
+
+    Fw::CmdResponse TraceFileLogger::process_traceId_storage(U32 traceId,bool enable) {
+        
+        //Add trace ID to the list (if it doesn't already exist) to stop logging it
+        bool IdProcessed = false;
+        if (enable == false) { 
+            if(this->filterTraceId.search_array(traceId, nullptr) == false) {
+                IdProcessed = this->filterTraceId.add_element(traceId);
+            }
+            else {
+                IdProcessed = true; //adding a duplicate element is ignored
+            }
+        }
+        else{
+            //Remove trace ID from list (if its in the list) to start logging it
+            IdProcessed = this->filterTraceId.delete_element(traceId);
+        }
+        if(IdProcessed == false) {
+            return Fw::CmdResponse::VALIDATION_ERROR;
+        }
+        else {
+            return Fw::CmdResponse::OK;
+        }
+    }
+
+    // ----------------------------------------------------------------------
+    // Handler implementations for user-defined typed input ports
+    // ----------------------------------------------------------------------
+
+    void TraceFileLogger::TraceBufferLogger_handler(FwIndexType portNum,
+                                                FwTraceIdType id,
+                                                Fw::Time& timeTag,
+                                                const Fw::TraceCfg::TraceType& type,
+                                                Fw::TraceBuffer& args) {
+            
+            FW_ASSERT(type.isValid());
+            U16 bit_mask = static_cast<U16> (FILTER_BIT << type.e);
+            U32 traceSize = 0;
+            
+            //Only log trace types that are enabled by either config or user
+            if(!(this->m_traceFilter & bit_mask)) {
+                //TODO: Should we generate an event here, letting user know that this specific filter is disabled?
+                return;
+            }
+            //Only log trace Ids that are not in the list 
+            if(this->filterTraceId.search_array(id,nullptr) == true) {
+                return;
+            }
+           
+           //Make a call to reset
+           Fw::SerializeBufferBase& buf_ref = m_file_buffer.getSerializeRepr();
+           buf_ref.resetSer();
+           (void)buf_ref.serialize(id);
+           (void)buf_ref.serialize(timeTag);
+           if(FW_TRACE_RECORD_MINIMAL == false) {
+            (void)buf_ref.serialize(args);
+            traceSize = m_file_buffer.getSize(); //Record max size of each trace record for circular file
```

Comment:
Not understanding this - why wouldn't it be `getBuffLength()` too?

Reply:
Because if it is minimal, it is already fixed size and you can store more records...unless you think it would cause issues with ground decoding?